src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -61,14 +64,15 @@ public DefaultCaseDataAccessControl(RoleAssignmentService roleAssignmentService,
             .filter(roleAssignments, caseDetails);
         CaseTypeDefinition caseTypeDefinition = caseTypeService.getCaseType(caseDetails.getCaseTypeId());
 
-        List<AccessProfile> accessProfiles = roleAssignmentToAccessProfileMapper
-            .toAccessProfiles(filteringResults, caseTypeDefinition);
-
         if (applicationParams.getEnablePseudoRoleAssignmentsGeneration()) {
             List<RoleAssignment> augmentedRoleAssignments = fakeRoleAssignmentsGenerator
                 .addFakeRoleAssignments(filteringResults);
         }
 
+        List<AccessProfile> accessProfiles = roleAssignmentToAccessProfileMapper
+            .toAccessProfiles(filteringResults, caseTypeDefinition);
+
+        List<AccessProfile> generatedAccessProfiles = roleToAccessProfilesMappingsGenerator.generate(caseTypeDefinition);
```

Comment:
the naming chosen is confusing. It's basically a similar naming to the line above. We are not generating the role to access profiles in the PR. That's what the line above does. You are augmenting the mapping by adding some pseudo mappings

Reply:
furthermore, there's a missing requirement specified in the ticket that hasn't been implemented. This time I'll leave to you to find out which one is that :)

Reply:
furthermore, you need to add the list generatedAccessProfiles to the already existing list of AccessProfiles generated by the previous step